src/coreclr/ToolBox/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -162,27 +149,101 @@ unsigned int MethodContext::saveToFile(HANDLE hFile)
 #define LWM(map, key, value) sparseWriteFileCR(map);
 #include "crlwmlist.h"
 
-    // Write the end canary
+    // Add canary
     buff2[buffIndex++] = '4';
     buff2[buffIndex++] = '2';
+    Assert(buffIndex == totalLen);
 
-    Assert(buffIndex == totalFileSize);
+    return data;
+}
 
-    WriteFile(hFile, buff2, totalFileSize, &bytesWritten, NULL);
-    delete[] buff2;
+static std::vector<unsigned char> applyCompression(const std::vector<unsigned char>&& data, MethodContextCompression compression)
```

Comment:
Shouldn't this be a `MethodContext` static (private) member? Same for the decompression function.